### PR TITLE
Closes #2361: add HttpHeaders constants

### LINKED ISSUE #2361: Implement constants for shared HTTP headers
It'd be nice if apps could share a single HTTP header constant definition with values like, "Content-Type" and "Authorization". A list of common HTTP request headers can be found at: https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Standard_request_fields

@pocmo What's a good place for this? I'd think:
- `support-base`
- `support-utils`
- A new HTTP abstraction module

I'm leaning towards base because it's a "building block". I think it should be separate from `concept-fetch` because, in theory, some consumers may not want to use our abstraction but might want pre-defined headers.

### PR DESCRIPTION
**Pull request by mcomella**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
  - I don't know what to test: they're string constants. To make sure no one accidentally changes them and breaks _all_ the dependent apps, we could add sanity check tests like, "CONTENT_TYPE" includes "Content-Type".
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by mcomella** [comment]
btw, I didn't add move all the existing content-type constants over because it's the end of my day and I ran out of time (it's a lot of work though too, to add the "support-base" dependencies, add the import statement, and replace the text).

File: components/support/base/src/main/java/mozilla/components/support/base/http/HttpHeaders.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * A list of common HTTP request headers can be found at
+ *   https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Standard_request_fields
+ *
+ * A list of common HTTP response headers can be found at
+ *   https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Standard_response_fields
+ */
+object HttpHeaders {
+
+    const val CONTENT_TYPE = "Content-Type"
+}
```
Comment (pocmo) [inline components/support/base/src/main/java/mozilla/components/support/base/http/HttpHeaders.kt]: I like the idea of those constants. But wouldn't we want to move them into `concept-fetch`?

Comment (mcomella) [reply components/support/base/src/main/java/mozilla/components/support/base/http/HttpHeaders.kt]: I discussed this (briefly) in the issue: https://github.com/mozilla-mobile/android-components/issues/2361#issue-421299227

**Comment by mcomella** [comment]
@pocmo I wasn't sure where to put the headers. Now that there is a `Headers` class already, it felt weird adding another one but with `Http` prefixing it, it sounds more confusing. I opted for `Headers.Common` but let me know if it should move elsewhere.

**Comment by mcomella** [comment]
@pocmo Another consideration we might have: if we want to include values for both common header names but also values (e.g. Content-Type header might have a value of `application/x-www-form-urlencoded`).

**Comment by mcomella** [comment]
I'm not sure how to fit values (e.g. ^) into the current model.

**Comment by pocmo** [comment]
@pocmo Another consideration we might have: if we want to include values for both common header names but also values (e.g. Content-Type header might have a value of application/x-www-form-urlencoded).

🤷‍♂️ 

Maybe on the `Body` since that is what has the type? :)

**Comment by mcomella** [comment]
Glean intermittent failure: rerunning.

**Comment by mcomella** [comment]
> Maybe on the `Body` since that is what has the type? :)

@pocmo `Request.Body`? But this is unrelated to the request: I'm referring to the "Header-Value" i n "Header-Name: Header-Value". Maybe `Headers.Common` and `Headers.Common.Values`?

It could also be `Headers.Names` and `Headers.Values`... 🤔

**Pull request merged**